File: dotnet/shared/Status/StatusModels.cs
using System.Text.Json.Serialization;

namespace ZyxxyzShared.Status;

public record ServiceStatus(
    [property: JsonPropertyName("id")]      string  Id,
    [property: JsonPropertyName("label")]   string  Label,
    [property: JsonPropertyName("ok")]      bool?   Ok,
    [property: JsonPropertyName("latency")] int?    Latency,
    [property: JsonPropertyName("ts")]      long?   Ts,
    [property: JsonPropertyName("uptime")]  double? Uptime);

public record SparklinePoint(
    [property: JsonPropertyName("ok")] bool Ok,
    [property: JsonPropertyName("ts")] long Ts);

public abstract record ApiResult<T>
{
    public sealed record Ok(T Value)      : ApiResult<T>;
    public sealed record Err(string Msg)  : ApiResult<T>;
}

File: kotlin/shared/src/commonMain/kotlin/xyz/zyxwonderland/shared/api/StatusModels.kt
package xyz.zyxwonderland.shared.api

import kotlinx.serialization.Serializable

@Serializable
data class ServiceStatus(
    val id: String,
    val label: String,
    val ok: Boolean?,
    val latency: Int?,
    val ts: Long?,
    val uptime: Double?,
)

@Serializable
data class SparklinePoint(
    val ok: Boolean,
    val ts: Long,
)

sealed class ApiResult<out T> {
    data class Success<T>(
        val data: T,
    ) : ApiResult<T>()

    data class Error(
        val message: String,
    ) : ApiResult<Nothing>()
}
